pkg/build/util/generate.go

```diff
@@ -13,6 +14,7 @@ import (
 // GenerateBuildFromConfig creates a new build based on a given BuildConfig. Optionally a SourceRevision for the new
 // build can be specified.  Also optionally a list of image names to be substituted can be supplied.  Values in the BuildConfig
 // that have a substitution provided will be replaced in the resulting Build
+// You NEED to update BuildConfig object after calling this!
```

Comment:
There is no atomic update you can do here, so you either rely on the build name being unique, or you don't try to do this.